### ISSUE TITLE
async callbacks

### ISSUE THREAD

**Issue by nikolasdas** [hywax/nuxt-cron#49]
Any plans for `defineCronHandler` to accept an async `CronTick` callback? Would be very helpful! :)

**Comment by hywax** [comment]
Now you can use asynchronous callbacks.

```ts
import { defineCronHandler } from '#nuxt/cron'

export default defineCronHandler('everyMinute', async () => {
  const data = await $fetch<{ repos: any[] }>('https://ungh.cc/orgs/unjs/repos')

  console.log(`UnJS has ${data.repos.length} repositories available`)
}, { runOnInit: true })

// stdOut: UnJS has 88 repositories available
```

If it doesn't make sense to you, write a case study.

**Comment by nikolasdas** [comment]
Sorry I should have clarified this a bit more, passing a async callback seems to work fine, yes.
But the type definition requires a sync function, would it be possible to change this?

**Comment by hywax** [comment]
Can you give me an example, I guess I don't understand what you're talking about.

**Comment by nikolasdas** [comment]
https://github.com/hywax/nuxt-cron/blob/b69fcc23e05fcf5146666a6cc5c3f9cc5ffe056f/src/runtime/types.d.ts#L24
Because of this type definition I get a `@typescript-eslint/no-misused-promises` warning. I know this might be nitpicky, but I try to have my eslint rules strict and avoid eslint-disable comments 🙈 
I can create a PR if you'd like, but wanted to make sure it is okay to use an async function in the first place

**Comment by hywax** [comment]
Looked at the types, I can replace `CronTick`

```ts
export type CronTick = () => void | Promise<void>
```

Am I correct in assuming that's what you're looking for?

**Comment by nikolasdas** [comment]
Yes, this would make my linter and subsequently me happier 👍

**Comment by hywax** [comment]
Fixed it in [v1.5.0](https://github.com/hywax/nuxt-cron/releases/tag/v1.5.0).

Thank you so much for your contribution!

**Comment by nikolasdas** [comment]
Thank you for your awesome support! And this cool nuxt module of course :)
